src/sentry/integrations/slack/webhooks/action.py

```diff
@@ -942,3 +700,222 @@ def handle_member_approval(self, slack_request: SlackActionRequest, action: str)
+    ) -> None:
+        # Error handling is done in the calling function
+        slack_client.views_open(
+            trigger_id=slack_request.data["trigger_id"],
+            view=modal_payload,
+        )
+
+    def open_dialog(self, slack_request: SlackActionRequest, group: Group) -> None:
+        # XXX(epurkhiser): In order to update the original message we have to
+        # keep track of the response_url in the callback_id. Definitely hacky,
+        # but seems like there's no other solutions [1]:
+        #
+        # [1]: https://stackoverflow.com/questions/46629852/update-a-bot-message-after-responding-to-a-slack-dialog#comment80795670_46629852
+        org = group.project.organization
+
+        callback_id_dict = {
+            "issue": group.id,
+            "orig_response_url": slack_request.data["response_url"],
+            "is_message": _is_message(slack_request.data),
+            "rule": slack_request.callback_data.get("rule"),
+        }
+
+        if slack_request.data.get("channel"):
+            callback_id_dict["channel_id"] = slack_request.data["channel"]["id"]
+        callback_id = orjson.dumps(callback_id_dict).decode()
+
+        # only add tags to metadata
+        metadata_dict = callback_id_dict.copy()
+        metadata_dict["tags"] = list(slack_request.get_tags())
+        metadata = orjson.dumps(metadata_dict).decode()
+
+        # XXX(CEO): the second you make a selection (without hitting Submit) it sends a slightly different request
+        modal_payload = self.get_modal_payload(callback_id, metadata=metadata)
+        slack_client = SlackSdkClient(integration_id=slack_request.integration.id)
+        try:
+            # We need to use the action_ts as the external_id to update the modal
+            # We passed this in control when we sent the loading modal to beat the 3 second timeout
+            external_id = slack_request.get_action_ts()
+
+            if options.get("send-slack-response-from-control-silo"):
+                self._update_modal(slack_client, external_id, modal_payload, slack_request)
+            else:
+                self._open_view(slack_client, modal_payload, slack_request)
+
+            metrics.incr(
+                SLACK_WEBHOOK_GROUP_ACTIONS_SUCCESS_DATADOG_METRIC,
+                sample_rate=1.0,
+                tags={"type": f"{self.dialog_type}_modal_open"},
+            )
+        except SlackApiError:
+            metrics.incr(
+                SLACK_WEBHOOK_GROUP_ACTIONS_FAILURE_DATADOG_METRIC,
+                sample_rate=1.0,
+                tags={"type": "resolve_modal_open"},
+            )
+            _logger.exception(
+                "slack.action.response-error",
+                extra={
+                    "organization_id": org.id,
+                    "integration_id": slack_request.integration.id,
+                    "trigger_id": slack_request.data["trigger_id"],
+                    "dialog": self.dialog_type,
+                },
+            )
```

Comment:
Is there a plan to have our other integrations implement a similar Modal flow as slack? I could see it being useful to wrap and raise these exceptions, letting the caller choose the response to make metric and logging decisions a bit more generic.

Reply:
No plans as yet, but that sounds like something we'll probably want to do as part of unifying our SLOs.

In another dev branch that's open right now, I'm already running into the problem of wanting to open an interaction recordk, but then not knowing whether the interaction succeeded or failed because exceptions are being caught and swallowed further down in the call stack.